File: it.unibo.boundaryWalk/app/src/main/java/it/unibo/annotatedExample/ISSActorSpec.java
package it.unibo.annotatedExample;

import java.lang.annotation.*;

@Target(value = { ElementType.CONSTRUCTOR, ElementType.METHOD, ElementType.TYPE })
@Retention(RetentionPolicy.RUNTIME)
@Inherited
public  @interface ISSActorSpec {
     //String sendMsg( );
     String actorName() default "unknown";   //annotations and interfaces can't have constructors

}

File: it.unibo.boundaryWalk/app/src/main/kotlin/it/unibo/annotExample/ISSActorSpec.kt
package it.unibo.annotExample

import java.lang.annotation.Inherited
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Target(AnnotationTarget.CONSTRUCTOR, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(RetentionPolicy.RUNTIME)
@Inherited
annotation class ISSActorSpec( //String sendMsg( );
    val actorName: String = "unknown" //annotations and interfaces can't have constructors
)